recirq/benchmarks/rep_rate/rep_rate_calculator.py

```diff
@@ -0,0 +1,263 @@
+        self.qubits = list(self.device.qubits)
+        if gate_set == cg.SYC_GATESET:
+            self.gate = cirq.google.SYC
+
+        # log of print statements for testing
+        self.print_log = ''
+
+    def _flush_print_log(self) -> None:
+        """Remove print log used fro debugging."""
+        self.print_log = ''
+
+    def _print(self, s: str) -> None:
+        """Includes printing to testing log and flushing."""
+        self.print_log += s
+        self.print_log += '\n'
+        print(s, flush=True)
+
+    def _latency_circuit(self) -> cirq.Circuit:
+        """Circuit for measuring round-trip latency.
+
+        Use the most simple circuit possible, a measurement of a single
+        random qubit.
+        """
+        return cirq.Circuit(cirq.measure(random.choice(self.qubits)))
+
+    def _entangling_layer(self, qubits: List[cirq.Qid]) -> cirq.Moment:
+        """Creates a layer of random two-qubit gates."""
+        m = cirq.Moment()
+        for q in qubits:
+            if q in m.qubits:
+                continue
+            pairings = [
+                q + adj
+                for adj in [(0, 1), (1, 0), (0, -1), (-1, 0)]
+                if q + adj in qubits and q + adj not in m.qubits
+            ]
+            if not pairings:
+                continue
+            q2 = random.choice(pairings)
+            m = m.with_operation(self.gate(q, q2))
+        return m
+
+    def _sq_layer(self, qubits: List[cirq.Qid], parameterized: bool,
+                  symbol_start: int) -> (cirq.Moment, int):
+        """Creates a layer of single-qubit gates.
+
+        If parameteried is true, this will add symbols to the qubits
+        in order to test parameter resolution.
+        """
+        m = cirq.Moment()
+        current_sym = symbol_start
+        for q in qubits:
+            if parameterized:
+                symbol = f's_{current_sym}'
+                current_sym += 1
+                m = m.with_operation(cirq.X(q)**sympy.Symbol(symbol))
+            else:
+                m = m.with_operation(
+                    cirq.PhasedXZGate(x_exponent=random.random(),
+                                      z_exponent=random.random(),
+                                      axis_phase_exponent=random.random())(q))
+        return (m, current_sym)
+
+    def _create_rep_rate_circuit(
+            self, parameterized: bool, width: int, depth: int,
+            num_sweeps: Optional[int]) -> Tuple[cirq.Circuit, cirq.Sweep]:
+        """Creates a testing circuit based on parameters.
+
+        The circuit will be of alternating single and two qubit layers
+        using a number of qubits specified by width and a number of
+        moments specified by depth.
+
+        This function will also create a sweep if parameterized is true,
+        by parameterized all single qubit layers with random values for
+        angles.
+        """
+        qubits = self.qubits[:width]
+        c = cirq.Circuit()
```

Comment:
would `cirq.experiments.random_quantum_circuit_generation` functions work here?

Reply:
Yeah, I am not sure.  I punted on that for now.  The tricky part would be for parameterizing the circuits, which I don't think this library can do.  I would definitely like to keep parameter sweeps in this, since it is a crucial component of rep rate (and we have had problems with parameterizing being slow before).